kandora.bot/utils/TensoulLogParser/TensoulParser.cs

```diff
@@ -0,0 +1,132 @@
+﻿using kandora.bot.http;
+using kandora.bot.models;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace kandora.bot.utils.TensoulLogParser;
+
+public static class TensoulParser
+{
+    public static RiichiGame ParseTensoulFormatGame(string payload, GameType gameType)
+    {
+        var game = JsonSerializer.Deserialize<TensoulGame>(payload);
+        var node = JsonNode.Parse(payload);
+        var riichiGame = game.ToRiichiGame(node.SelectByPath("$.log"));
+        riichiGame.GameType = gameType;
+
+        return riichiGame;
+    }
+
+    public static RiichiGame ToRiichiGame(this TensoulGame tenhouGame, JsonNode resultGame)
+    {
+        return new RiichiGame()
+        {
+            Dan = tenhouGame.Dans,
+            Ref = tenhouGame.Reference,
+            Ver = tenhouGame.Version,
+            Rule = tenhouGame.Rule.ToRiichiRule(),
+            FinalRankDeltas = tenhouGame.Scores.GetRanks(),
```

Comment:
let's use one-liners here using linq, like:
`return sc.Select(item => (int)item).ToArray();`
since it's the only place we do that

Reply:
Changed both FinalRankDeltas and FinalScores to Linq expressions :D